System/DataModule.cpp

```diff
@@ -1,764 +1,497 @@
+			// Find the type entry in the type map
+			map<std::string, std::list<std::pair<std::string, Entity *> > >::iterator clsItr = m_TypeMap.find(withType);
+			if (clsItr == m_TypeMap.end()) { return false; }
+
+			const std::list<std::string> *pGroupList = 0;
+			// Go through all the entities of that type, adding the groups they belong to
+			for (std::list<std::pair<std::string, Entity *> >::iterator instItr = clsItr->second.begin(); instItr != clsItr->second.end(); ++instItr) {
+				pGroupList = instItr->second->GetGroupList();
+				// Get the grouped entities, without transferring ownership
+				for (std::list<std::string>::const_iterator gItr = pGroupList->begin(); gItr != pGroupList->end(); ++gItr) {
+					groupList.push_back(*gItr);
+					foundAny = true;
+				}
+			}
+			// Make sure there are no dupe groups in the list
+			groupList.sort();
+			groupList.unique();
+		}
+		return foundAny;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	bool DataModule::GetAllOfGroup(std::list<Entity *> &entityList, std::string group, std::string type) {
+		if (group.empty()) { return false; }
+
+		bool foundAny = false;
+
+		// Look in all classes, so only look in the Entity level, which includes all!
+		if (type.empty() || type == "All") {
+			map<std::string, std::list<std::pair<std::string, Entity *> > >::iterator clsItr = m_TypeMap.find("Entity");
+			if (clsItr == m_TypeMap.end()) { return false; }
+
+			// Go through only the Entity typelist since it should contain all entities in this DataModule
+			for (std::list<std::pair<std::string, Entity *> >::iterator instItr = clsItr->second.begin(); instItr != clsItr->second.end(); ++instItr) {
+				// Get the grouped entities, without transferring ownership
+				if (instItr->second->IsInGroup(group)) {
+					entityList.push_back(instItr->second);
+					foundAny = true;
+				}
+			}
+		} else {
+			// Look only in one specific class (which will get all derived classes' entities too!)
+			// Look for the class entry
+			map<std::string, std::list<std::pair<std::string, Entity *> > >::iterator clsItr = m_TypeMap.find(type);
+			if (clsItr == m_TypeMap.end()) { return 0; }
+
+			RTEAssert(!clsItr->second.empty(), "DataModule has class entry without instances in its map!?");
+
+			for (std::list<std::pair<std::string, Entity *> >::iterator instItr = clsItr->second.begin(); instItr != clsItr->second.end(); ++instItr) {
+				// Get the grouped entities, without transferring ownership
+				if (instItr->second->IsInGroup(group)) {
+					entityList.push_back(instItr->second);
+					foundAny = true;
+				}
+			}
+		}
+		return foundAny;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	bool DataModule::GetAllOfType(std::list<Entity *> &entityList, std::string type) {
+		if (type.empty()) { return false; }
```

Comment:
Early return here and below.
Spaces between `>`s
All the comments except maybe the one about not transferring ownership are unnecessary and made clear by the code.